examples/get-insights-data/main.py

```diff
@@ -0,0 +1,121 @@
+import os
+import csv
+from utils.utils import PublicAPIClient, get_commandline_arguments
+
+
+def main():
+    args = get_commandline_arguments(
+        [
+            (
+                ["course_id"],
+                {"type": int, "help": "id of the course to get insights for"},
+            )
+        ]
+    )
+    public_api_client = PublicAPIClient(args.subdomain)
+    perspectives = fetch_course_perspectives(public_api_client, args.course_id)
+    course_data = fetch_course_data(public_api_client, args.course_id)
+
+    construct_summary(public_api_client, perspectives, course_data)
+    construct_student_insights(public_api_client, perspectives, course_data)
+
+
+def fetch_course_perspectives(public_api_client, course_id):
+    response = public_api_client.request("get", f"courses/{course_id}/perspectives")
+    return response.json()["perspectives"]
+
+
+def fetch_course_data(public_api_client, course_id):
+    response = public_api_client.request("get", f"courses/{course_id}")
+    return response.json()["course"]
+
+
+def construct_summary(public_api_client, perspectives, course):
+    headers = [
+        "Course ID",
+        "Course Title",
+        "Perspective UUID",
+        "Perspective Title",
+        "Views",
+        "Time Viewed [min]",
+        "Unique Viewers",
+    ]
+    csv_data = []
+    for perspective in perspectives:
+        perspective_data = [
+            course["id"],
```

Comment:
This should be `course['course_id']` here and everywhere (altogether 4 places) in this script. That's the Canvas ID.

Reply:
true, changed